src/traces/bar/hover.js

```diff
@@ -65,17 +65,18 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
     if(Color.opacity(mc)) pointData.color = mc;
     else if(Color.opacity(mlc) && mlw) pointData.color = mlc;
 
+    var size = (trace.base) ? di.b + di.s : di.s;
```

Comment:
The bars of traces that set `base` have their `di.b` set accordingly and are excluded from stacking. The size of these bars is computed as `di.b + di.s`.

The bars of traces that don't set `base` have their `di.b` set to 0, unless they are stacked, in which case, the stacking algorithm sets `di.b`. For these bars, the bar size is `di.s`; no need to add `di.b` (either because it'd be 0, or because it'd be something that the stacking algorithm has set to stack the bars).